src/Identity/Core/src/Passkeys/PasskeyAssertionState.cs

```diff
@@ -0,0 +1,13 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Identity;
+
+// Represents the state to persist between creating the passkey request options
+// and performing passkey assertion.
+internal sealed class PasskeyAssertionState
+{
+    public required ReadOnlyMemory<byte> Challenge { get; init; }
+
+    public string? UserId { get; init; }
+}
```

Comment:
Prior to this PR, we were persisting the entire passkey creation/request options JSON strings in the auth cookie and using them during the attestation and assertion procedures. With these changes, we only store the state we absolutely need to, and we rely on global configuration (in `PasskeyOptions`) for everything else. This new type defines which state gets persisted during the assertion procedure (options creation + credential validation), and `PasskeyAttestationState` is similar but for attestation.